src/lib/operators/get_table.cpp

```diff
@@ -12,82 +12,162 @@
+        deleted_chunk_ids.emplace_back(chunk_id);
       }
     }
   }
 
-  if (temp_excluded_chunk_ids.empty()) {
-    return original_table;
+  /**
+   * Early out if no pruning of Chunks or Columns is necessary
+   */
+  if (deleted_chunk_ids.empty() && _pruned_chunk_ids.empty() && _pruned_column_ids.empty()) {
+    return stored_table;
   }
 
-  // We create a copy of the original table, but omit excluded chunks
-  const auto pruned_table = std::make_shared<Table>(original_table->column_definitions(), TableType::Data,
-                                                    original_table->max_chunk_size(), original_table->has_mvcc());
+  // We cannot create a Table without columns - since Chunks rely on their first column to determine their row count
+  Assert(_pruned_column_ids.size() < stored_table->column_count(), "Cannot prune all columns from Table");
+  DebugAssert(std::all_of(_pruned_column_ids.begin(), _pruned_column_ids.end(),
+                          [&](const auto column_id) { return column_id < stored_table->column_count(); }),
+              "ColumnID out of range");
+
+  /**
+   * Build pruned TableColumnDefinitions of the output Table
+   */
+  auto pruned_column_definitions = TableColumnDefinitions{};
+  if (_pruned_column_ids.empty()) {
+    pruned_column_definitions = stored_table->column_definitions();
+  } else {
+    pruned_column_definitions =
+        TableColumnDefinitions{stored_table->column_definitions().size() - _pruned_column_ids.size()};
+
+    auto pruned_column_ids_iter = _pruned_column_ids.begin();
+    for (auto stored_column_id = ColumnID{0}, output_column_id = ColumnID{0};
+         stored_column_id < stored_table->column_count(); ++stored_column_id) {
+      if (pruned_column_ids_iter != _pruned_column_ids.end() && stored_column_id == *pruned_column_ids_iter) {
+        ++pruned_column_ids_iter;
+        continue;
+      }
 
-  std::sort(temp_excluded_chunk_ids.begin(), temp_excluded_chunk_ids.end());
-  temp_excluded_chunk_ids.erase(std::unique(temp_excluded_chunk_ids.begin(), temp_excluded_chunk_ids.end()),
-                                temp_excluded_chunk_ids.end());
+      pruned_column_definitions[output_column_id] = stored_table->column_definitions()[stored_column_id];
+      ++output_column_id;
+    }
+  }
+
+  /**
+   * Build the output Table, omitting pruned Chunks and Columns as well as deleted Chunks
+   */
+  const auto output_table = std::make_shared<Table>(pruned_column_definitions, TableType::Data,
+                                                    stored_table->max_chunk_size(), stored_table->has_mvcc());
+
+  auto pruned_chunk_ids_iter = _pruned_chunk_ids.begin();
+  auto deleted_chunk_ids_iter = deleted_chunk_ids.begin();
+
+  for (ChunkID chunk_id{0}; chunk_id < stored_table->chunk_count(); ++chunk_id) {
+    // Exclude the Chunk if is either pruned or deleted.
+    // A Chunk can be both, so we have to make sure both iterators are incremented in that case
```

Comment:
> You initialized the _deleted_chunk_ids vector with the _pruned_chunk_ids. Hence, a pruned chunk would always be a deleted chunk. On purpose?

Whops, that was indeed a mistake!

> Also, I would like more info about the iterators. Don't get the comment.

We cannot just do

```c++
if (pruned_chunk_ids_iter != _pruned_chunk_ids.end() && *pruned_chunk_ids_iter == chunk_id) {
  ++pruned_chunk_ids_iter;
 continue;
}
```

because then `deleted_chunk_ids_iter` might still point to the ChunkID that we just skipped due to `pruned_chunk_ids_iter`

Reply:
"Either or" is to me an xor. But that's not the case here. I think that was my initial point of confusion. I thought there is more behind it as there apparently is.

Reply:
Removed the "either"